manual/src/tutorials/effectexamples.etex

```diff
@@ -0,0 +1,666 @@
+\chapter{Effect Handlers} \label{c:effects-tutorial}
+%HEVEA\cutname{effects-tutorial.html}
+
+Effect handlers are a mechanism for modular programming with user-defined
+effects. Effect handlers allow the programmers to describe
+\textit{computations} that \textit{perform} effectful \textit{operations}, whose
+meaning is described by \textit{handlers} that enclose the computations. Effect
+handlers are a generalization of exception handlers, and enable non-local
+control-flow mechanisms such as resumable exceptions, lightweight threads,
+coroutines, generators and asynchronous I/O to be composably expressed. In this
+tutorial, we shall see how some of these mechanisms can be built using effect
+handlers.
+
+\section{s:effects-basics}{Basics}
+
+To understand the basics, let us define an effect that exchanges an integer,
+and a computation that performs two exchanges and returns their sum.
+
+\begin{caml_example*}{verbatim}
+open Effect
+open Effect.Deep
+
+type _ Effect.t += Xchg: int -> int t
+let comp1 () = perform (Xchg 0) + perform (Xchg 1)
+\end{caml_example*}
+
+We declare the exchange effect "Xchg" by extending the pre-defined extensible
+variant type "Effect.t" with a new constructor "Xchg: int -> int t". The
+declaration may be intuitively read as ``the "Xchg" effect take an integer
+parameter, and when this effect is performed, it returns an integer''. The
+computation "comp1" performs the effect twice using the "perform" primitive and
+returns their sum.
+
+In this example, we use the deep version of the effect handlers here as opposed
+to shallow version. The distinction between deep and shallow handlers is
```

Comment:
"shallow version" should be "the shallow version". Also, I would suggest adding a one-line explanation of the difference between shallow and deep handlers, and an indication of which is simpler / more common. E.g., "A deep handler monitors a computation until the computation terminates (either normally or via an exception), and handles all of the effects performed by the computation. In contrast, a shallow handler monitors a computation until either the computation terminates or the computation performs one effect, and it handles this single effect only. In situations where they are applicable, deep handlers are usually preferred."